bindings/nodejs/src/structures/sol-js-gpio.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <nan.h>
+#include "sol-js-gpio.h"
+#include "../common.h"
+#include "handles.h"
+
+using namespace v8;
+
+static void sol_gpio_read_callback(void *data, struct sol_gpio *gpio, bool value) {
+    Nan::HandleScope scope;
+    Nan::Callback *callback = (Nan::Callback *)data;
+    if (!callback)
+        return;
+
+    Local<Value> arguments[2] = {
+        js_sol_gpio(gpio),
+        Nan::New(value)
+    };
+    callback->Call(2, arguments);
+}
+
+bool c_sol_gpio_config(v8::Local<v8::Object> jsGPIOConfig, sol_gpio_config *p_config) {
+    sol_gpio_config config = {
+        SOL_GPIO_CONFIG_API_VERSION,
```

Comment:
While I don't expect someone using JS to disable Ali version to save memory, the good practice is to isolate in ifdef or use the SOL_SET_API() macro (I'm on my phone, so the name may be slightly different)